File: food-ordering-system/restaurant-service/src/main/java/mk/finki/restaurantservice/infrastructure/RestaurantRepositoryAdapter.java
package mk.finki.restaurantservice.infrastructure;

import lombok.RequiredArgsConstructor;
import mk.finki.restaurantservice.domain.Restaurant;
import mk.finki.restaurantservice.domain.RestaurantRepository;
import org.springframework.stereotype.Component;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Component
@RequiredArgsConstructor
public class RestaurantRepositoryAdapter implements RestaurantRepository {
    private final RestaurantJpaRepository jpaRepository;

    @Override
    public Optional<Restaurant> findById(UUID id) {
        return jpaRepository.findById(id);
    }

    @Override
    public List<Restaurant> findAll() {
        return jpaRepository.findAll();
    }

    @Override
    public Restaurant save(Restaurant restaurant) {
        return jpaRepository.save(restaurant);
    }
}

File: food-ordering-system/restaurant-service/src/main/kotlin/mk/finki/restaurantservice/infrastructure/persistence/RestaurantRepositoryAdapter.kt
package mk.finki.restaurantservice.infrastructure.persistence

import mk.finki.restaurantservice.domain.restaurant.Restaurant
import mk.finki.restaurantservice.domain.restaurant.RestaurantRepository
import mk.finki.restaurantservice.domain.restaurant.valueobjects.RestaurantId
import org.springframework.stereotype.Component
import java.util.*

@Component
class RestaurantRepositoryAdapter(
    private val jpaRepository: RestaurantJpaRepository
) : RestaurantRepository {
    override fun findById(id: RestaurantId): Optional<Restaurant> = jpaRepository.findById(id)
    override fun findAll(): List<Restaurant> = jpaRepository.findAll()
    override fun save(restaurant: Restaurant): Restaurant = jpaRepository.save(restaurant)
}
